openmc/universe.py

```diff
@@ -1117,3 +1178,64 @@ def _partial_deepcopy(self):
         clone.auto_geom_ids = self.auto_geom_ids
         clone.auto_mat_ids = self.auto_mat_ids
         return clone
+
+    def add_cell(self, cell):
+        """Add a cell to the universe.
+
+        Parameters
+        ----------
+        cell : openmc.Cell
+            Cell to add
+
+        """
+
+        if not isinstance(cell, openmc.Cell):
+            msg = f'Unable to add a Cell to Universe ID="{self._id}" since ' \
+                  f'"{cell}" is not a Cell'
+            raise TypeError(msg)
+
+        cell_id = cell.id
+
+        if cell_id not in self._cells:
+            self._cells[cell_id] = cell
+
+    def add_cells(self, cells):
+        """Add multiple cells to the universe.
+
+        Parameters
+        ----------
+        cells : Iterable of openmc.Cell
+            Cells to add
+
+        """
+
+        if not isinstance(cells, Iterable):
+            msg = f'Unable to add Cells to Universe ID="{self._id}" since ' \
+                  f'"{cells}" is not iterable'
+            raise TypeError(msg)
+
+        for cell in cells:
+            self.add_cell(cell)
+
+    def remove_cell(self, cell):
+        """Remove a cell from the universe.
+
+        Parameters
+        ----------
+        cell : openmc.Cell
+            Cell to remove
+
+        """
+
+        if not isinstance(cell, openmc.Cell):
+            msg = f'Unable to remove a Cell from Universe ID="{self._id}" ' \
+                  f'since "{cell}" is not a Cell'
+            raise TypeError(msg)
+
+        # If the Cell is in the Universe's list of Cells, delete it
+        self._cells.pop(cell.id, None)
+
+    def clear_cells(self):
```

Comment:
This is alread provided on `UniverseBase`. Is there a reason it is duplicated here?

Reply:
This is part of `Universe`, not `UniverseBase` directly.
As `DAGMCUniverse` inherit from `UniverseBase` I as unsure of the best implementation.

I can move all of them in `UniverseBase` so they both inherit from the methods.

Reply:
You're right, they aren't inherited methods on the `Universe` class, but if we're implementing the same method in two places it would be good to consider whether or not they could be placed on the `UniverseBase` class, even as abstract methods so we have a consistent interface.